code/weapon/weapons.cpp

```diff
@@ -5764,10 +5777,8 @@ int weapon_create( vec3d * pos, matrix * porient, int weapon_type, int parent_ob
 
 		objp->radius = model_get_radius(wip->model_num);
 
-		// if we intrinsic-rotate, make sure we have a model instance
-		if (model_get(wip->model_num)->flags & PM_FLAG_HAS_INTRINSIC_ROTATE) {
-			wp->model_instance_num = model_create_instance(false, wip->model_num);
-		}
+		// Always create an instance in case we need them
+		wp->model_instance_num = model_create_instance(false, wip->model_num);
```

Comment:
I don't like this; the vast majority of weapons are not going to need a model instance.  Can this be created conditionally?

Reply:
The issue is that the action code only deals with model instances since that is the way to go now. Making this conditional would require a lot of special handling for something that (as far I know) only requires a bit more memory allocations.

Reply:
I don't think it would require too much special handling.  All that's necessary is to check whether the instance is -1, something that's done in the relevant places (such as the rendering code) already.  Also, I just looked through the PR and there are only two places that need model instances, neither of which are going to be used for weapons.  If you revert these few lines, the rest of the PR will work with no further changes.

Yes, it just requires a bit of extra memory, but there are a *lot* of weapons used in every mission.  It adds up.

Reply:
Model instances are required for weapons if the action moves between subobjects of a weapon.

Something I could do relatively easily would be to check if we have a program and only create the instance in that case. However, that seems like a weird design to me since it would require changes for every program that is added which I would like to avoid to keep the code as generic as possible.

Would something along those lines be acceptable?

Reply:
My thinking is - how many programs are going to need to operate on the weapon's subobjects rather than the weapon as a hole?  The only uses I can think of are very esoteric, and I would rather cross that bridge when we come to it.

However, your suggestion is an acceptable alternative.